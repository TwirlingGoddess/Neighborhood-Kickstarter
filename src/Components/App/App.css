.App {
  text-align: center;
  background-image: url('../../images/neighborhood-back.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
}

h1,
h2,
h3,
h4,
h5 {
  font-family: 'Space Mono', monospace;
}

