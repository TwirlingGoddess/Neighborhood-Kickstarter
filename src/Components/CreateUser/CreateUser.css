.create-user-section {
  align-items: center;
  display: flex;
  height: 100vh;
  width: 100%;
  justify-content: center;
}

h1 {
  color: white;
}

.create-user-form {
  display: flex;
  align-items: center;
  border-radius: 0.7em;
  background-image: linear-gradient(to top, #09203f 0%, #537895 100%);
  justify-content: flex-start;
  width: 30rem;
  height: 28rem;
  flex-direction: column;
}

.input-fields {
  margin: .3rem;
  color: black;
  padding-left: .4rem;
  width: 20rem;
  height: 2.5rem;
  font-size: 1.5rem;
  background-image: linear-gradient(-225deg, #FFFEFF 0%, #D7FFFE 100%);
  border-radius: 0.6em;
  border: none;
}

.select-input {
  width: 20.2rem;
  margin-top: .5rem;
  font-size: .7rem;
  border-radius: 0;
}

.sign-up-button {
  box-sizing: border-box;
  appearance: none;
  background-color: transparent;
  border-radius: 0.6em;
  cursor: pointer;
  align-self: center;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1;
  margin-top: 25px;
  padding: 1em 1em;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 700;
  border-color: #3498db;
  color: #fff;
  box-shadow: 0 0 40px 40px #3498db inset, 0 0 0 0 #3498db;
  transition: all 150ms ease-in-out;
  width: 19.8rem;
  max-height: 4rem;
}

.sign-up-button:hover {
  color: #fff;
  outline: 0;
  box-shadow: 0 0 10px 0 #3498db inset, 0 0 10px 4px #3498db;
}