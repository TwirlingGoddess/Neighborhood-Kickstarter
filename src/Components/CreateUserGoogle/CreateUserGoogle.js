import React, { Component } from 'react';
import Select from 'react-select';
import GoogleLogin from 'react-google-login';

import { getNeighborhoods, addNewUser } from '../../utilities/apiCalls/apiCalls'

import './CreateUserGoogle.css'

class CreateUserGoogle extends Component {
  constructor() {
    super()
    this.state = {
      userName: '',
      neighborhood: '',
      neighborhoods: []
    }
  }

  componentDidMount = () => {
    this.setNeighborhoods();
  }

  responseGoogle = async (response) => {
    let { updateUser } = this.props;
    let googleSignIn = {
      first_name: response.profileObj.givenName,
      last_name: response.profileObj.familyName,
      email: response.profileObj.email,
      username: this.state.userName,
      district_id: this.state.neighborhood.id,
      token: response.profileObj.googleId
    };
    
    const user = await addNewUser(googleSignIn);

    if (user) {
      updateUser(user);
      this.props.history.push('/Landing')
    }
  }

  setNeighborhoods = async() => {
    const response = await getNeighborhoods();

    const neighborhoods = response.map(neighborhood => {
      return { value: neighborhood.name, label: neighborhood.name, id: neighborhood.id }
    })

    this.setState({
      neighborhoods
    })
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    })
  }

  handleSelectChange = (selectedOption) => {
    const neighborhood = selectedOption;
    this.setState({
      neighborhood
    })
  }

  handleSubmit = (event) => {
    event.preventDefault();

    this.setState({
      userName: '',
      neighborhood: ''
    })

    this.props.history.push('/Landing')
  }

  render() {
    return (
      <div className='create-user-section'>
        <form className='create-user-form-google' onSubmit={this.handleSubmit}>
          <h1>Create Account</h1>
          <input
            className='input-fields-google'
            name='userName'
            type='text'
            value={this.state.userName}
            placeholder='User Name'
            onChange={this.handleChange}
          />
          <Select
            className='select-input' 
            placeholder='Select your neighborhood'
            value={this.state.neighborhood}
            onChange={this.handleSelectChange}
            options={this.state.neighborhoods}
          />
          <GoogleLogin
            clientId="603748791729-1qgv1pg7tl426jut42re2tnub34nu0hu.apps.googleusercontent.com"
            buttonText="Login with Google"
            onClick={this.responseGoogle}
            disabled={this.state.neighborhood && this.state.userName ? false : true}
            onSuccess={this.responseGoogle}
            onFailure={this.responseGoogle}
          />
        </form>
      </div>
    )
  }
}

export default CreateUserGoogle