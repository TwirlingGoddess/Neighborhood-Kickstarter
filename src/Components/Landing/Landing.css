.landing-page {
  margin: auto;
  width: 100vw;
  height: 100vh;
  /*overflow: scroll;*/
}