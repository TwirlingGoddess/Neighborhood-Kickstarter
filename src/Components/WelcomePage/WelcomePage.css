.user-link-section {
  display: flex;
  width: 100%;
  height: 100vh;
  justify-content: center;
}

.user-link {
  box-sizing: border-box;
  appearance: none;
  background-color: transparent;
  border-radius: 0.6em;
  cursor: pointer;
  align-self: center;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1;
  margin: 20px;
  padding: 1em 1em;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 700;
  border-color: #3498db;
  color: #fff;
  box-shadow: 0 0 40px 40px #3498db inset, 0 0 0 0 #3498db;
  transition: all 150ms ease-in-out;
  width: 12rem;
  max-height: 4rem;
}

.user-link:hover {
  color: #fff;
  outline: 0;
  box-shadow: 0 0 10px 0 #3498db inset, 0 0 10px 4px #3498db;
}

.signin {
  height: 5rem;
}