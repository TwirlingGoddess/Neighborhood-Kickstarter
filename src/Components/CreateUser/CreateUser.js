import React, { Component } from 'react';
import Select from 'react-select';
import { getNeighborhoods, addNewUserLocal, sendEmailConfirmation } from '../../utilities/apiCalls/apiCalls';
import './CreateUser.css';
import PropTypes from 'prop-types';

class CreateUser extends Component {
  constructor() {
    super();
    this.state = {
      allUsers: [],
      firstName: '',
      lastName: '',
      userName: '',
      email: '',
      password: '',
      neighborhood: '',
      neighborhoods: []
    };
  }

  componentDidMount = () => {
    this.setNeighborhoods();
  }

  setNeighborhoods = async() => {
    const response = await getNeighborhoods();

    const neighborhoods = response.map(neighborhood => {
      return { value: neighborhood.name, label: neighborhood.name, id: neighborhood.id };
    });

    this.setState({
      neighborhoods
    });
  }

  handleChange = (event) => {
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  }

  handleSelectChange = (selectedOption) => {
    const neighborhood = selectedOption;

    this.setState({
      neighborhood
    });
  }

  handleSubmit = async (event) => {
    event.preventDefault();

    let {firstName, lastName, email, password, neighborhood, userName} = this.state;
    let { updateUser } = this.props;

    let localSignIn = {
      first_name: firstName,
      last_name: lastName,
      email,
      username: userName,
      district_id: neighborhood.id,
      password
    };

    let user = await addNewUserLocal(localSignIn);

    if (user) {
      let emailUser = {user_name: user.first_name, user_email: user.email};

      await sendEmailConfirmation(emailUser);

      updateUser(user);
      
      this.props.history.push('/Landing');
    }
  }

  render() {
    return (
      <div className='create-user-section'>
        <form className='create-user-form' onSubmit={this.handleSubmit}>
          <h1>Create Account</h1>
          <input
            className='input-fields first-name-input'
            name='firstName'
            type='text'
            value={this.state.firstName}
            placeholder='first name'
            onChange={this.handleChange}
          />
          <input 
            className='input-fields last-name-input'
            name='lastName'
            type='text'
            value={this.state.lastName}
            placeholder='last name'
            onChange={this.handleChange}
          />
          <input 
            className='input-fields'
            name='userName'
            type='text'
            value={this.state.userName}
            placeholder='user name'
            onChange={this.handleChange}
          />
          <input 
            className='input-fields email-input'
            name='email'
            type='email'
            value={this.state.email}
            placeholder='email'
            onChange={this.handleChange}
          />
          <input
            className='input-fields'
            name='password'
            type='text'
            value={this.state.password}
            placeholder='password'
            onChange={this.handleChange}
          />
          <Select
            className='select-input' 
            placeholder='Select your neighborhood'
            value={this.state.neighborhood}
            onChange={this.handleSelectChange}
            options={this.state.neighborhoods}
          />
          <button className='sign-up-button'>Sign Up</button>
        </form>
      </div>
    );
  }
}

CreateUser.propTypes = {
  history: PropTypes.array
};

export default CreateUser;