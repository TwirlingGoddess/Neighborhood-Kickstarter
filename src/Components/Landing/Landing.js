import React, { Component } from 'react';
import Search from'../Search/Search';
import { ProjectsContainer } from '../ProjectsContainer/ProjectsContainer';
import './Landing.css'

class Landing extends Component {
  constructor() {
    super()
    this.state = {
      currentUser: ''
    }
  }
  
  render() {
    return(
      <div className='landing-page'>
        <h1>Landing</h1>
        <Search />
        <ProjectsContainer />
      </div>
    )
  }
}

export default Landing