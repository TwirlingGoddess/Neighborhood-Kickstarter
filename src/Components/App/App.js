import React, { Component } from 'react';
import './App.css';
import { Route, withRouter } from 'react-router-dom';
import WelcomePage from '../WelcomePage/WelcomePage';
import CreateUser from '../CreateUser/CreateUser';
import SignIn from '../SignIn/SignIn'
import Landing from '../Landing/Landing';
import CreateProject from '../CreateProject/CreateProject'
import CreateUserGoogle from '../CreateUserGoogle/CreateUserGoogle';
import UserProjects from '../UserProjects/UserProjects';
import { GoogleLogin } from 'react-google-login';

class App extends Component {
  constructor() {
    super();
    this.state = {
      currentUser: {}
    };
  }

  updateUser = (user) => {
    this.setState({
      currentUser: user
    }) 
  }

  logOutUser = () => {
    this.setState({
      currentUser: {}
    })
  }

  render() {
    const currentUser = this.state.currentUser;
    return (
      <div className="App">
         <Route exact path= '/' component={WelcomePage} />
         <Route exact path= '/CreateUser' component={CreateUser} />
         <Route exact path = '/CreateUserGoogle' render={({history}) => 
           <CreateUserGoogle updateUser={this.updateUser} history={history}/>
         } />
         <Route exact path= '/Landing' render={() => 
           <Landing currentUser={currentUser} logOutUser={this.logOutUser}/>
         } />
         <Route exaxt path= '/SignIn' render={({history}) => 
          <SignIn currentUser={currentUser} updateUser={this.updateUser} history={history}/>
         } />
         <Route exact path= '/CreateProject' render={() => 
          <CreateProject currentUser={currentUser}/>
        }/>
        <Route exact path= '/UserProjects' render={() => 
          <UserProjects currentUser={currentUser}/>
        }/>
      </div>
    );
  }
}

export default withRouter(App);
