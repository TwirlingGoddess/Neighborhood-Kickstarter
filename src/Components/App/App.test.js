import React from 'react';
import App from './App';
import { shallow, mount } from 'enzyme';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router';
import WelcomePage from '../WelcomePage/WelcomePage';
import CreateUser from '../CreateUser/CreateUser';
import CreateUserGoogle from '../CreateUserGoogle/CreateUserGoogle';
import Landing from '../Landing/Landing';

describe('Contributions', () => {
  let wrapper;
  let initial;
  let expected;
  let mockUserState = {};
  let mockProjectState = {};
  let mockEvent = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a valid path to the Landing', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/' ]}>
        <App/>
      </MemoryRouter>
    );
    expect(wrapper.find(WelcomePage)).toHaveLength(1);
  });

  it('should have a valid path to the Landing', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/CreateUser' ]}>
        <App/>
      </MemoryRouter>
    );
    expect(wrapper.find(CreateUser)).toHaveLength(1);
  });

  it('should have a valid path to the Landing', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/CreateUserGoogle' ]}>
        <App/>
      </MemoryRouter>
    );
    expect(wrapper.find(CreateUserGoogle)).toHaveLength(1);
  });

  it('should have a valid path to the Landing', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/Landing' ]}>
        <App/>
      </MemoryRouter>
    );
    expect(wrapper.find(Landing)).toHaveLength(1);
  });  

  // it.only('Should have an initial currentUser state of an empty object', () => {
  //   expected = {};
  //   expect(wrapper.state('currentUser')).toEqual(expected)
  // })

  // it('Should update the currentUser in state', () => {
  //   wrapper = shallow(<App updateUser={jest.fn}/>);
  //   wrapper.instance()

  // })

})
