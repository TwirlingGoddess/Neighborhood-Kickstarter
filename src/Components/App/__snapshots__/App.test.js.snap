// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contributions Should update the state when logOutUser is invoked 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(App)
    logOutUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Contributions Should update the state when updateUser is invoked 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(App)
    updateUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Contributions should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(App) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
